src/chart_types/xy_chart/domains/x_domain.ts

```diff
@@ -127,6 +123,24 @@ export function mergeXDomain(
   };
 }
 
+function getMinInterval(computedMinInterval: number, size: number, customMinInterval?: number) {
+  if (customMinInterval != null) {
+    // Allow greater custom min if xValues has 1 member.
+    if (size > 1 && customMinInterval > computedMinInterval) {
+      Logger.warn(
+        'custom xDomain is invalid, custom minInterval is greater than computed minInterval. Using computed minInterval.',
+      );
+      return computedMinInterval;
+    }
+    if (customMinInterval < 0) {
+      Logger.warn('custom xDomain is invalid, custom minInterval is less than 0. Using computed minInterval.');
+      return computedMinInterval;
+    }
+  }
+
+  return customMinInterval || computedMinInterval;
+}
```

Comment:
this can also be cleaned up a bit
```suggestion
  if (customMinInterval == null) {
    return computedMinInternval;
  }
    // Allow greater custom min if xValues has 1 member.
    if (size > 1 && customMinInterval > computedMinInterval) {
      Logger.warn(
        'custom xDomain is invalid, custom minInterval is greater than computed minInterval. Using computed minInterval.',
      );
      return computedMinInterval;
    }
    if (customMinInterval < 0) {
      Logger.warn('custom xDomain is invalid, custom minInterval is less than 0. Using computed minInterval.');
      return computedMinInterval;
    }
  return customMinInterval || computedMinInterval;
}
```

Reply:
See 3c7c2f805b4d9a163b285da552b9d59848e75461